cmd/lifecycle/analyzer.go

```diff
@@ -77,21 +76,26 @@ func (a *analyzeCmd) Args(nargs int, args []string) error {
 	if nargs != 1 {
 		return cmd.FailErrCode(fmt.Errorf("received %d arguments, but expected 1", nargs), cmd.CodeInvalidArgs, "parse arguments")
 	}
+
 	if args[0] == "" {
 		return cmd.FailErrCode(errors.New("image argument is required"), cmd.CodeInvalidArgs, "parse arguments")
 	}
-	a.imageName = args[0]
-
-	if a.previousImage == "" {
-		a.previousImage = a.imageName
-	}
+	a.outputImageRef = args[0]
 
 	if a.restoresLayerMetadata() {
-		if a.platform06.cacheImageTag == "" && a.platform06.cacheDir == "" {
+		if a.cacheImageRef == "" && a.platform06.cacheDir == "" {
 			cmd.DefaultLogger.Warn("Not restoring cached layer metadata, no cache flag specified.")
 		}
 	}
 
+	if a.previousImageRef == "" {
+		a.previousImageRef = a.outputImageRef
+	}
+
+	if err := image.ValidateDestinationTags(a.useDaemon, append(a.additionalTags, a.outputImageRef)...); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "validate image tag(s)")
+	}
+
```

Comment:
This is something that the exporter currently does. I wonder if we want to move it here in the spirit of failing fast.

Reply:
The spec has this wording for exporter:

> If `<daemon>` is false and more than one `<image>` is provided they MUST refer to the same registry

This could be added to the associated spec PR.

Reply:
I like this!

Reply:
Added to spec PR: https://github.com/buildpacks/spec/pull/234/commits/5718cbad8c3bd21f5f0637a72f0bdf70ef19bb8a